tests/e2e/performance/test_pvc_multi_clone_performance.py

```diff
@@ -0,0 +1,412 @@
+
+        self.params["clone_yaml"] = constants.CSI_CEPHFS_PVC_CLONE_YAML
+        if self.interface == constants.CEPHBLOCKPOOL:
+            self.params["clone_yaml"] = constants.CSI_RBD_PVC_CLONE_YAML
+
+        output = self.run_oc_command(
+            cmd=f"get pod {self.pod_obj.name} -o yaml", namespace=self.params["nspace"]
+        )
+
+        results = yaml.safe_load("\n".join(output))
+        self.params["path"] = results["spec"]["containers"][0]["volumeMounts"][0][
+            "mountPath"
+        ]
+        log.info(f"path - {self.params['path']}")
+
+        fd, tmpfile = tempfile.mkstemp(suffix=".yaml", prefix="Clone")
+        self.params["tmpfile"] = tmpfile
+
+        # reading template of clone yaml file
+        with open(self.params["clone_yaml"], "r") as stream:
+            try:
+                clone_yaml = yaml.safe_load(stream)
+                clone_yaml["spec"]["storageClassName"] = self.pvc_obj.backed_sc
+                clone_yaml["spec"]["dataSource"]["name"] = self.params["pvcname"]
+                clone_yaml["spec"]["resources"]["requests"]["storage"] = (
+                    str(self.pvc_obj.size) + "Gi"
+                )
+            except yaml.YAMLError as exc:
+                log.error(f"Can not read template yaml file {exc}")
+        log.info(
+            f'Clone yaml file : {self.params["clone_yaml"]} '
+            f"Content of clone yaml file {clone_yaml}"
+        )
+
+        return clone_yaml
+
+    def get_log_names(self):
+        """
+        Finds names for log files pods in which logs for clone creation are located
+        For CephFS: 2 pods that start with "csi-cephfsplugin-provisioner" prefix
+        For RBD: 2 pods that start with "csi-rbdplugin-provisioner" prefix
+
+        """
+        self.params["logs"] = []
+
+        pods = self.run_oc_command(cmd="get pod", namespace="openshift-storage")
+        if self.params["ERRMSG"] in pods:
+            raise Exception("Can not get csi controller pod")
+
+        provisioning_name = "csi-cephfsplugin-provisioner"
+        if self.interface == constants.CEPHBLOCKPOOL:
+            provisioning_name = "csi-rbdplugin-provisioner"
+
+        for line in pods:
+            if provisioning_name in line:
+                self.params["logs"].append(line.split()[0])
+        log.info(f'The logs pods are : {self.params["logs"]}')
+
+    def run_fio_on_pod(self, file_size):
+        """
+        Args:
+        file_size (str): Size of file to write in MB, e.g. 200M or 50000M
+
+        """
+        file_name = self.pod_obj.name
+        log.info(f"Starting IO on the POD {self.pod_obj.name}")
+        # Going to run only write IO to fill the PVC for the before creating a clone
+        self.pod_obj.fillup_fs(size=file_size, fio_filename=file_name)
+
+        # Wait for fio to finish - non default timeout is needed for big pvc/clones
+        fio_result = self.pod_obj.get_fio_results(timeout=18000)
```

Comment:
I'm concerned about using large timeout values. 5 hours for IO failure (although, we may hit other errors if there is a failure) sounds like a huge value. can we reduce this to 30 minutes?

Reply:
I will try it later with smaller timeout. Currently let's have it 30 mins and in the next PR I'll reduce it after validating the new timeout value.